nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/ConsumeKinesisStream.java

```diff
@@ -652,7 +652,9 @@ synchronized Scheduler prepareScheduler(final ProcessContext context, final Proc
         final LeaseManagementConfig leaseManagementConfig = configsBuilder.leaseManagementConfig()
                 .failoverTimeMillis(getFailoverTimeMillis(context))
                 .streamName(streamName);
-        final CoordinatorConfig coordinatorConfig = configsBuilder.coordinatorConfig().workerStateChangeListener(workerState::set);
+        final CoordinatorConfig coordinatorConfig = configsBuilder.coordinatorConfig()
+                .workerStateChangeListener(workerState::set)
+                .clientVersionConfig(CoordinatorConfig.ClientVersionConfig.CLIENT_VERSION_CONFIG_COMPATIBLE_WITH_2X);
```

Comment:
Is there a way to check the current status of the Kinesis stream to determine whether this compatibility flag is needed?

Reply:
It should be possible but I don't have the bandwidth now to investigate